src/ts/state/reducer.ts

```diff
@@ -124,21 +139,41 @@ export interface IPluginState<TMatches extends IMatch = IMatch> {
   };
   // The current error message.
   requestErrors: IMatchRequestError[];
+  // The current state of the filter
+  filterState: TFilterState;
 }
 
 // The transaction meta key that namespaces our actions.
 export const PROSEMIRROR_TYPERIGHTER_ACTION = "PROSEMIRROR_TYPERIGHTER_ACTION";
 
+interface IInitialStateOpts<
+  TFilterState extends unknown,
+  TMatch extends IMatch
+> {
+  doc: Node;
+  matches: TMatch[];
+  ignoreMatch: IIgnoreMatchPredicate;
+  matchColours: IMatchTypeToColourMap;
+  filterOptions: IFilterOptions<TFilterState, TMatch> | undefined;
```

Comment:
this: you must pass a value for `filterOptions`, even if it's undefined
`?`: you may optionally pass a value for `filterOptions`, but if you pass something, it must be defined

We use this here to allow us to transparently pass a possibly undefined value. If we didn't, we'd need to conditionally call the function with/without the argument depending on what the consumer code passed.

Having said that, the signature is a bit odd, eh? Maybe there's a better pattern here – I will have written this without thinking too hard to preserve the ergonomics in the calling code.

Reply:
Super interesting. In that case you might like to supply a default value of undef in the function below as per the other fields.

Reply:
Ah yeah, that's what I'm after!

Reply:
Brief correction – I glanced at the definition above assuming it was part of the function definition, but it isn't – it's the definition of the options object that's passed to the function.

As a result there's no opportunity to supply a default in the interface, and supplying a default of `undefined` in the function implementation is a no-op, because the object is destructured when it's received. So this probably doesn't need a change as-is.